agent/proxycfg/testing.go

```diff
@@ -1119,6 +1119,73 @@ func TestConfigSnapshotExposeConfig(t testing.T) *ConfigSnapshot {
 	}
 }
 
+func TestConfigSnapshotTerminatingGateway(t testing.T) *ConfigSnapshot {
+	return testConfigSnapshotTerminatingGateway(t, true)
+}
+
+func TestConfigSnapshotTerminatingGatewayNoServices(t testing.T) *ConfigSnapshot {
+	return testConfigSnapshotTerminatingGateway(t, false)
+}
+
+func testConfigSnapshotTerminatingGateway(t testing.T, populateServices bool) *ConfigSnapshot {
+	roots, _ := TestCerts(t)
+
+	snap := &ConfigSnapshot{
+		Kind:    structs.ServiceKindTerminatingGateway,
+		Service: "terminating-gateway",
+		ProxyID: structs.NewServiceID("terminating-gateway", nil),
+		Address: "1.2.3.4",
+		TaggedAddresses: map[string]structs.ServiceAddress{
+			structs.TaggedAddressWAN: structs.ServiceAddress{
+				Address: "198.18.0.1",
+				Port:    443,
+			},
+		},
+		Port:       8443,
+		Roots:      roots,
+		Datacenter: "dc1",
+	}
+	if populateServices {
+		web := structs.NewServiceID("web", nil)
+		webNodes := TestUpstreamNodes(t)
+		webNodes[0].Service.Meta = map[string]string{
+			"version": "1",
+		}
+		webNodes[1].Service.Meta = map[string]string{
+			"version": "2",
+		}
+
+		api := structs.NewServiceID("api", nil)
+		apiNodes := TestUpstreamNodes(t)
+		for i := 0; i < len(apiNodes); i++ {
+			apiNodes[i].Service.Service = "api"
+			apiNodes[i].Service.Port = 8081
+		}
+
+		// Hard-coding these certs since TestLeafForCA returns a new leaf each time and will not match the golden file
```

Comment:
I think you could probably use the golden TLS files, like so: https://github.com/hashicorp/consul/blob/da893c36a17f39002e6f4abac669c45155ee8541/agent/xds/clusters_test.go#L355-L364

Reply:
Done!